### ISSUE TITLE
Reorganize documentation

### ISSUE THREAD

**Issue by carolynvs** [getporter/porter#1513]
Split the documentation site into a couple sections based on the intended reader:

* End User installing bundles
* Bundle Author
* Mixin or Plugin Author

**Comment by DARK-art108** [comment]
@carolynvs is this still open! Like we are updating our docs to the new version so is it is required now!

**Comment by carolynvs** [comment]
@DARK-art108 Yes, it's still open and I have been slowly working on it. I have a [reorganized document structure](https://hackmd.io/roa3Nfh4SruG0FFKyJwb7Q?view) that I think would work better for us.

We have another open issue, #1644, to split up the site's doc into multiple versions (so that people can see docs relevant to the major version of Porter that they are working on)
